src/compiler/checker.ts

```diff
@@ -3760,22 +3888,24 @@ namespace ts {
         }
 
         function getTypeListId(types: Type[]) {
-            switch (types.length) {
-                case 1:
-                    return "" + types[0].id;
-                case 2:
-                    return types[0].id + "," + types[1].id;
-                default:
-                    let result = "";
-                    for (let i = 0; i < types.length; i++) {
-                        if (i > 0) {
-                            result += ",";
+            if (types) {

```

Comment:
Because a type argument list may now be empty (i.e. undefined) in a type reference. There's a comment on the `TypeReference` interface in types.ts that explains how it works.